### PR TITLE
feat: Add Repository secret observation

### LINKED ISSUE #36: Repository controller doesn't update ressource on password change
### What happened?

The [isUpToDate method](https://github.com/crossplane-contrib/provider-argocd/blob/18c39395956db553ddf7f5b303ed860c99e0edd0/pkg/controller/repositories/controller.go#L382) does not check if the password provided via secret changed.
The workaround is currently to trigger an update-call by changing another field in the repo claim.

### How can we reproduce it?

Deploy secret with e.g. token, deploy repo referencing the secret (see example). 
Write a new token in the secret or reference another secret with a new token. 
-> no update call/request + new token is not being used

### PR DESCRIPTION
**Pull request by oskisk**
### Description of your changes 

Trigger a resource update on secret changes for all the references within the repository controller.

Fixes #36

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

- Added basic unit tests for the repository controllers
- manual testing for referenes and observations

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by oskisk** [comment]
@MisterMX Would you have time to review this PR?

**Review by MisterMX — CHANGES_REQUESTED** [review]
Thanks for this PR @oskisk! I made some minor remarks. Can you take a look at them?

File: pkg/controller/repositories/controller.go [hunk 1]
```diff
@@ -114,34 +114,55 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		Repo: meta.GetExternalName(cr),
 	}
 
-	// workaround for https://github.com/argoproj/argo-cd/issues/5951
-	// we have to use ListRepositories() because Get() doesn't work -^
-	repository := &argocdv1alpha1.Repository{}
-	repositoryList, err := e.client.ListRepositories(ctx, &repoQuery)
+	observerdRepository, err := e.client.Get(ctx, &repoQuery)
```
[the remark this answers is not in the document]
Comment (oskisk) [reply pkg/controller/repositories/controller.go]: Fixed the typo, thx.

File: pkg/controller/repositories/controller.go [hunk 2]
```diff
@@ -114,34 +114,55 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
...
-	if repositoryList.Items != nil {
-		for _, r := range repositoryList.Items {
-			if r.Repo == repoQuery.Repo {
-				repository = r
-				break
-			}
+		if repositories.IsErrorRepositoryNotFound(err) {
+			return managed.ExternalObservation{
+				ResourceExists: false,
+			}, nil
```
[the remark this answers is not in the document]
Comment (oskisk) [reply pkg/controller/repositories/controller.go]: Thank you, I have changed it to your suggestion.

**Comment by oskisk** [comment]
> Thanks for this PR @oskisk! I made some minor remarks. Can you take a look at them?

Thank you for reviewing. I've made the changes to the branch based on your suggestions.

**Review by MisterMX — APPROVED** [review]
LGTM. Thank you very much @oskisk!

**Pull request merged**